backend/app/analysis/parts_of_speech.py

```diff
@@ -209,6 +205,26 @@ def get_word_examples(word_list, part_of_speech, text):
                 for example in net.examples():
                     if word.lower() in example.lower():
                         word_examples[word].append(example.lower())
-        else:
-            word_examples[word] = ''
+
+    sentences = re.split("[?.!]", text)
+    for sentence in sentences:
+        for word in word_list:
+            if word in sentence:
+                word_examples[word].append(sentence)
     return word_examples
+
+
+def punct_in_word(word):
+    """
+    Checks if there are punctuations in the word
+    """
+    quotes = ["“", '"', "'", "’", ".", "?", "!"]
+    for quote in quotes:
+        if quote in word:
+            return True
+    return False
+
+
+def get_valid_words(text, pos):
+    return list(set(word.lower() for word in get_part_of_speech_words(text.lower(), pos)
```

Comment:
This list comprehension and then casting list-set-list to ensure all elements are unique is pretty hard to read. Spelling out what's going on will make this easier to maintain:

```
pos_words = get_part_of_speech_words(text.lower(), pos)
valid_words = [
    word.lower() for word in pos_words
    if not punct_in_word(word) and len(word) > 2
]
unique_valid_words = list(set(valid_words))
return unique_valid_words
```

Might consider moving all of this validation into `get_part_of_speech_words` if it's always called inside this function.